pkg/framework/framework.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package framework
+
+import (
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/transport"
+	errors "golang.org/x/xerrors"
+)
+
+// AriesFramework provides access to clients being managed by the Framework.
+type AriesFramework struct {
+	opts options
+}
+
+type options struct {
+	OutboundTransport transport.OutboundTransport
+}
+
+// Option configures the framework.
+type Option func(opts *options) error
+
+// New initializes the Framework based on the set of options provided.
+func New(opts ...Option) (*AriesFramework, error) {
+	// get the default framework configs
+	def := defFramework{}
```

Comment:
You could alternatively represent defFramework as, itself, an Option that is processed first.

Reply:
[Assigned](https://github.com/hyperledger/aries-framework-go/compare/9b194c723afdf936279c58a665113ec88f3239e0..94efeb32066ee1f587c67c8642c5e2ff588c5efa#diff-7b2448fdc2181299533bc944239b9744R29) defaults directly to options.